### DISCUSSION TITLE
Templates or Parameters for Library Panels

### DISCUSSION THREAD

**Discussion by Holocene** [grafana/grafana#37313] [Dashboards, Panels & Visualizations]
Hi,

Thanks very much for interesting the new Library Panels feature. It seems very useful.

I wondered if you had any plans to support templating/parameters for library panels?

For example, I have the following two panels on one of my dashboards:

![image](https://user-images.githubusercontent.com/2182425/127363890-1de9c27f-2537-481b-a402-b82acb0ac4c7.png)

I would like to be able to make a single library panel that could be used to generate both of these panels, one instantiated with the parameters `(OnDemand, Yellow)`, and one with the parameters `(WhatIf, Purple)`. My dashboards often have lots of very similar panels that I copy+paste to generate - this makes refactoring them to be quite tedious. A templating/parameterizing system based on library panels would be very handy.

Is this something you've thought about implementing?

Thanks a bunch,
Oliver.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
